core/common/test/files/SmokeFileTest.kt

```diff
@@ -6,51 +6,255 @@
+        }
+    }
+
+    @Test
+    fun deleteFile() {
+        val p = createTempPath()
+        assertFailsWith<FileNotFoundException> {
+            FileSystem.System.delete(p, mustExist = true)
+        }
+        // Should not fail
+        FileSystem.System.delete(p, false)
+    }
+
+    @Test
+    fun moveNonExistingFile() {
+        assertFailsWith<FileNotFoundException> {
+            FileSystem.System.atomicMove(createTempPath(), createTempPath())
+        }
+    }
+
+    @Test
+    fun createDirectories() {
+        val p = Path(createTempPath(), "a", "b", "c")
+        FileSystem.System.createDirectories(p)
+        assertTrue(FileSystem.System.exists(p))
+
+        assertFailsWith<IOException> {
+            FileSystem.System.createDirectories(p, true)
+        }
+        FileSystem.System.createDirectories(p, false)
+
+        val p1 = Path(p, "d")
+        FileSystem.System.createDirectories(p1)
+        assertTrue(FileSystem.System.exists(p1))
+
+        var pr = p1
+        for (i in 0..3) {
+            FileSystem.System.delete(pr)
+            pr = pr.parent!!
+        }
+    }
+
+    @Test
+    fun pathParent() {
+        val p = Path(Path.separator.toString(), "a", "b", "c")
+        assertEquals(constructAbsolutePath("a", "b"), p.parent?.toString())
+        assertEquals(constructAbsolutePath("a"), p.parent?.parent?.toString())
+        assertEquals(constructAbsolutePath(), p.parent?.parent?.parent?.toString())
+        assertNull(p.parent?.parent?.parent?.parent)
+
+        val p1 = Path("home", "..", "lib")
+        assertEquals(constructRelativePath("home", ".."), p1.parent?.toString())
+        assertEquals("home", p1.parent?.parent?.toString())
+        assertNull(p1.parent?.parent?.parent)
+
+        assertNull(Path("").parent)
+        assertNull(Path(".").parent)
+        assertNull(Path("..").parent)
+        assertNull(Path(Path.separator.toString()).parent)
+
+        assertEquals("..", Path("..${Path.separator}..").parent?.toString())
+    }
+
+    @Test
+    fun pathConcat() {
+        assertEquals(
+            constructAbsolutePath("a", "b", "c"),
+            Path(Path(Path(Path(Path.separator.toString()), "a"), "b"), "c").toString()
+        )
+
+        assertEquals(
+            constructAbsolutePath("a", "b", "..", "c"),
+            Path("${Path.separator}a", "b", "..${Path.separator}c").toString()
+        )
+
+        assertEquals(constructRelativePath("a", "b", "c"),
+            Path("", "a", "b", "c").toString())
+    }
+
+    @Test
+    fun fileName() {
+        assertEquals("", Path("").name)
+        assertEquals("hello", Path("hello").name)
+        assertEquals("", Path(Path.separator.toString()).name)
+        assertEquals(".", Path(".").name)
+        assertEquals("..", Path("..").name)
+        assertEquals("hello.txt", Path("base", "hello.txt").name)
+        assertEquals("dir", Path("dir${Path.separator}").name)
+    }
+
+    @Test
+    fun isAbsolute() {
```

Comment:
Could you please elaborate on why some tests are named `testFoo`, while the others are just `foo`?

No strong opinions here

Reply:
Just a lack of strong opinion on test method names, I guess. I'll use `foo` instead of `testFoo` here, thanks for pointing to it!

Reply:
Fixed